src/plugins/maps_legacy/public/index.ts

```diff
@@ -49,13 +53,32 @@ export {
   colorUtil,
   convertToGeoJson,
   IServiceSettings,
-  KibanaMap,
   KibanaMapLayer,
   VectorLayer,
   FileLayerField,
   FileLayer,
   TmsLayer,
 };
 
+// Due to a leaflet/leaflet-draw bug, it's not possible to consume leaflet maps w/ draw control
+// through a pipeline leveraging angular. For this reason, client plugins need to
+// init kibana map and the basemaps visualization directly rather than consume through
+// the usual plugin interface
+export function getKibanaMapFactoryProvider(core: CoreSetup) {
+  bindSetupCoreAndPlugins(core);
+  return (...args: any) => new KibanaMap(...args);
+}
+
+export function getBaseMapsVis(core: CoreSetup) {
+  const getKibanaMap = getKibanaMapFactoryProvider(core);
+  const serviceSettings = new ServiceSettings();
```

Comment:
Don't we need to make sure the serviceSettings get correctly initialized with min/max zoom/license?

Reply:
Good feedback! We were definitely missing the license here. Since it was a new/different instance of `ServiceSettings`, `maps_legacy_licensing` never injected the license in. I've defaulted instead to all plugins working with the same `ServiceSettings` instance as it was before.

I haven't modified how min/max zoom are set in this PR but that could be related to the above issue you mentioned where tiles aren't appearing  past zoom level 10. This issue also exists on master and should be tackled as part of a separate fix. I've captured it here: https://github.com/elastic/kibana/issues/64573